src/common/signalk/SKJSONVisitor.cpp

```diff
@@ -30,16 +30,47 @@
 
 #include "SKJSONVisitor.h"
 
+void SKJSONVisitor::processSource(const SKSource &source, JsonObject &sourceObject) {
+  sourceObject["label"] = source.getLabel().c_str();
+
+  switch (source.getInput()) {
```

Comment:
@neftaly Thanks for following the PRs!

The way it's done right now you will get the exact input on which data was received (NMEA2000, NMEA1, or NMEA2). 

Internal sensors like IMU, Barometer and ADCs will appear as coming from KBox. 

What kind of extra precision are you looking for? Can you give me an example?